File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        nums_map: dict[int] = {}

        
        for num in nums:
            nums_map[num] = nums_map.get(num, 0) + 1

        return heapq.nlargest(k, nums_map.keys(), key=nums_map.get)

File: Data Structures & Algorithms/duplicate-integer/submission-1.rs
impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut dup_set = HashSet::new();

        for num in nums{

            if dup_set.contains(&num){
                return true;
            }
            dup_set.insert(num);
        }
        false
    }
}
